2-Intelligent-Agents/c_cleaningRobot.js

```diff
@@ -233,7 +234,144 @@ function makeTableControlledDiagram() {
+        let percept = diagram_agent1.world.floors[location].dirty;
+        let table = getRulesFromPage();
+        let action = reflexVacuumAgent(diagram_agent1.world, table);
+        setTimeout(function(){  if(action=='SUCK'){ score_agent1 = score_agent1+50; }
+                                else{ score_agent1 = score_agent1-10; }
+                                if(diagram_agent1.world.floors[0].dirty || diagram_agent1.world.floors[1].dirty)
+                                  { score_agent1 = score_agent1-5; }
+                                performance_agent1.push(score_agent1);      
+                                diagram_agent1.world.simulate(action);        
+                                renderWorld(diagram_agent1);
+                                renderAgentPercept(diagram_agent1, percept);
+                                renderAgentAction(diagram_agent1, action);},table[1]*1000);
+    }
+    
+    //update agent2's environment
+    function update_agent2() {
+        let location = diagram_agent2.world.location;
+        let percept = diagram_agent2.world.floors[location].dirty;
+        let table = getRulesFromPage();
+        let action = reflexVacuumAgent(diagram_agent2.world, table);
+        setTimeout(function(){  if(action=='SUCK'){ score_agent2 = score_agent2+50; }
+                                else{ score_agent2 = score_agent2-10; }
+                                if(diagram_agent2.world.floors[0].dirty || diagram_agent2.world.floors[1].dirty)
+                                  { score_agent2 = score_agent2-5-parseInt(parameters[0]); }
+                                // extra score reduction due to agent2 working for a shorter duration
+                                performance_agent2.push(score_agent2);
+                                diagram_agent2.world.simulate(action);
+                                renderWorld(diagram_agent2);
+                                renderAgentPercept(diagram_agent2, percept);
+                                renderAgentAction(diagram_agent2, action);},table[1]*1000);
+        }
+
+    //print performace scores
+    //setInterval(plotPerformance(performance_agent1, performance_agent2),STEP_TIME_MS);
+    setInterval(function(){ plotPerformance(performance_agent1, performance_agent2); },STEP_TIME_MS);
+
+    //get reader defined parameters
+    function getRulesFromPage() {
+        let table = d3.select("#performance-controlled-diagram table");
+        let dirt_freq = table.select("[data-action=dirt-freq] select").node().value;
+        let speed_agent1 = table.select("[data-action=speed-agent1] select").node().value;
+        let speed_agent2 = table.select("[data-action=speed-agent2] select").node().value;
+        let interval_agent2 = table.select("[data-action=interval-agent2] select").node().value;
+        return [dirt_freq, speed_agent1, speed_agent2, interval_agent2]
+    }
+
+    function makefloordirty() {
+        floorNumber = Math.floor(Math.random() * 2);
```

Comment:
Is this a local or a global?

Reply:
I made some mistakes while setting the scopes of the variable. I will correct it with other changes and be more careful in the future.